mapie/conformity_scores/sets/aps.py

```diff
@@ -0,0 +1,438 @@
+from typing import Optional, Tuple, Union, cast
+
+import numpy as np
+from sklearn.dummy import check_random_state
+from sklearn.calibration import label_binarize
+
+from mapie.conformity_scores.sets.naive import Naive
+from mapie.conformity_scores.sets.utils import (
+    check_include_last_label, check_proba_normalized
+)
+from mapie.estimator.classifier import EnsembleClassifier
+
+from mapie._machine_precision import EPSILON
+from mapie._typing import ArrayLike, NDArray
+from mapie.utils import compute_quantiles
+
+
+class APS(Naive):
+    """
+    Adaptive Prediction Sets (APS) method-based non-conformity score.
+    It is based on the sum of the softmax outputs of the labels until the true
+    label is reached, on the calibration set. See [1] for more details.
+
+    References
+    ----------
+    [1] Yaniv Romano, Matteo Sesia and Emmanuel J. Candès.
+    "Classification with Valid and Adaptive Coverage."
+    NeurIPS 202 (spotlight) 2020.
+
+    Attributes
+    ----------
+    classes: Optional[ArrayLike]
+        Names of the classes.
+
+    random_state: Optional[Union[int, RandomState]]
+        Pseudo random number generator state.
+
+    quantiles_: ArrayLike of shape (n_alpha)
+        The quantiles estimated from ``get_sets`` method.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def get_predictions(
```

Comment:
where is this method used ?

Reply:
mapie/conformity_scores/classification.py:l155 in `get_sets` method